File: spotlight/src/main/java/com/takusemba/spotlight/OnSpotlightListener.java
package com.takusemba.spotlight;

import com.takusemba.spotlight.target.Target;

/**
 * internal listener to notify when something happened on a {@link Target}
 */
interface OnSpotlightListener {

  /**
   * called when SpotlightView is clicked
   */
  void onSpotlightViewClicked();
}

File: spotlight/src/main/java/com/takusemba/spotlight/OnSpotlightListener.kt
package com.takusemba.spotlight

/**
 * On Spotlight Ended Listener
 */
interface OnSpotlightListener {

  /**
   * Called when Spotlight is started
   */
  fun onStarted()

  /**
   * Called when Spotlight is ended
   */
  fun onEnded()
}
